packages/story-editor/src/app/story/utils/deleteNestedKeys.ts

```diff
@@ -39,7 +36,14 @@ function deleteNestedKeys(paths) {
       }
       const lastKey = keys.pop();
       const nextLastKey = keys.pop();
-      const nextLastObj = keys.reduce((a, key) => a?.[key] || a, object);
+      const nextLastObj = keys.reduce<Partial<T>>(
+        (a, key) => a?.[key] || a,
+        object
+      );
+
+      if (!nextLastObj || !nextLastKey || !lastKey) {
+        return;
+      }
       // Make sure we're not trying to get a property out of `undefined` or `null`.
       if (
         Object.prototype.hasOwnProperty.call(nextLastObj, nextLastKey) &&
```

Comment:
@barklund This is the other TS error that I haven't figure out yet (see line 53 below, can't add a comment there):
`delete nextLastObj[nextLastKey][lastKey];`

Getting `Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.`

Possible that the generic for `deleteNestedKeys` is just not correct, not sure what would be correct either though.

Reply:
For this one, I would rewrite the function to specifically remove keys on the resource object, and type it as exactly that. It's a mess to type this one correctly or even come close, and we don't actually need to. If we ever need to remote other keys that just properties directly on the resource, we can re-examine this. I do see that we sometimes need properties two-levels deep, but it could potentially just be two functions? If we really want to, I can look into typing this one properly, but it's not gonna be pretty 😂.

Reply:
Currently this util supports any object, there might be more props there causing unwanted history entries, I believe we even had an issue for it somewhere (couldn't find it currently). Would be great if we wouldn't have to change the logic to accommodate types.

So if you can think of anything, let's type properly 😛 Alternatively, we could skip this one line and create a follow-up issue. Thoughts?